presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -92,6 +92,62 @@ Additionally, the relations within a ``WITH`` clause can chain::
     relation is used. This means that if the relation is used more than once and the query
     is non-deterministic, the results may be different each time.
 
+
+WITH RECURSIVE Clause
+---------------------
+
+The ``WITH RECURSIVE`` clause is a variant of the ``WITH`` clause. It defines a
+``UNION`` of two relations. The first relation is called the *recursion base*,
+and the second relation is called the *recursion step*:
+
+.. code-block:: none
+
+    WITH RECURSIVE T(n) AS (
+    <recursion base relation>
+    UNION (ALL | DISTINCT)
+    <recursion step relation: SELECT .. FROM ... T...>
+    )
+    ... SELECT ... FROM T
```

Comment:
Following what I said before, this is not a full form of `WITH RECURSIVE` clause. I think it should be mentioned. This is the simplest and, probably, the most common form - it only has a single query in the list.

Also, please clean up the multiple dots and surrounding spaces in this code block.

Reply:
I am not sure how you want this cleaned up. Also could we actually make it an example that replaces <recursion base relation> ..